src/item.cpp

```diff
@@ -3651,16 +3651,9 @@ int item::get_encumber_when_containing( const units::volume &contents_volume ) c
 
 layer_level item::get_layer() const
 {
-    if( has_flag( "SKINTIGHT" ) ) {
-        return UNDERWEAR;
-    } else if( has_flag( "WAIST" ) ) {
-        return WAIST_LAYER;
-    } else if( has_flag( "OUTER" ) ) {
-        return OUTER_LAYER;
-    } else if( has_flag( "BELTED" ) ) {
-        return BELTED_LAYER;
-    }
-    return REGULAR_LAYER;
+    // We assume that an item will never have per-item flags defining its
+    // layer, so we can defer to the itype.
```

Comment:
`item::has_flag` also checks the flags of the contained gunmods, and the gunmod "shoulder_strap` has the flag "BELTED", which makes the whole gun that contains it have that flag.

Reply:
> `item::has_flag` also checks the flags of the contained gunmods, and the gunmod "shoulder_strap` has the flag "BELTED", which makes the whole gun that contains it have that flag.

Ah, good catch.  Do you think it would be correct to use the precomputed value when itype is armour, and fall back to the old algorithm otherwise?